packages/cli/src/UserManagement/UserManagementHelper.ts

```diff
@@ -45,6 +45,23 @@ export async function saveCredentialOwnership(
 	})) as SharedCredentials;
 }
 
+export async function getWorkflowOwner(workflowId: string) {
+	const workflowDb = await Db.collections.Workflow!.findOneOrFail(workflowId, {
+		relations: ['shared', 'shared.user', 'shared.user.globalRole'],
+	});
+
+	return workflowDb.shared[0].user;
+}
```

Comment:
A workflow can be shared with multiple users. Do we have a convention that the zeroth user it is shared with is considered the workflow's "owner"? Maybe it makes no difference, but I would expect an `owner` to be defined only with reference to an instance.

Reply:
Also, most calls to `getWorkflowOwner` stringify the arg, but I think TypeORM is lenient on string/number IDs, so it should still work if you set `workflowId: string | number` and simplify the calls.

Reply:
Changed the function to accept string or number for workflow ID and started query from shared workflows table, making it simpler to execute.

In regards to making it ready for sharing, actually this is a huge change because all runtime checks will be different then, and will have to receive a list of users instead of a single user.

Due to this reason, we are following the premise that each workflow contains one and only one owner.